pkg/proxy/endpoints.go

```diff
@@ -140,9 +140,12 @@ func (ect *EndpointChangeTracker) Update(previous, current *v1.Endpoints) bool {
 		change.previous = ect.endpointsToEndpointsMap(previous)
 		ect.items[namespacedName] = change
 	}
-	if t := getLastChangeTriggerTime(endpoints); !t.IsZero() {
-		ect.lastChangeTriggerTimes[namespacedName] =
-			append(ect.lastChangeTriggerTimes[namespacedName], t)
+	// TODO(#81360): ignore case when Endpoint is deleted.
+	if current != nil {
```

Comment:
Instead of this if, I would prefer the following:
```
if t := getlastChangeTriggerTime(current); ... {
}
```
[so just change endpoints to current here]

And the in getLastChangeTriggerTime handle the nil case:
```
if endpoints == nil {
  # TODO(#81360): Figure out how to handle Endpoints deletion
  return time.Time{}
}
```